parquet/src/main/java/org/apache/iceberg/parquet/ParquetReadSupport.java

```diff
@@ -57,7 +58,7 @@ public ReadContext init(Configuration configuration, Map<String, String> keyValu
 
     MessageType projection = ParquetSchemaUtil.hasIds(fileSchema) ?
         ParquetSchemaUtil.pruneColumns(fileSchema, expectedSchema) :
-        ParquetSchemaUtil.pruneColumnsFallback(fileSchema, expectedSchema);
+        ParquetSchemaUtil.pruneColumnsByName(fileSchema, expectedSchema, MappingUtil.create(expectedSchema));
```

Comment:
This should not replace the existing fallback strategy. If the mapping is present, it should be used. Otherwise, the existing position-based fallback strategy should be used.

Reply:
Agreed.